nvflare/private/fed/app/simulator/simulator_runner.py

```diff
@@ -207,6 +207,12 @@ def setup(self):
                 self.logger.error("Please provide the number of threads or provide gpu options to run the simulator.")
                 return False
 
+            if meta.get(JobMetaKey.MIN_CLIENTS, 0) > len(self.client_names):
+                self.logger.error(
+                    "The number of simulating clients can not be smaller than the minimum number of clients specified in job's meta."
+                )
+                return False
```

Comment:
I am not sure if we should enforce this. The main purpose of the Simulator is to help the researcher develop / debug the FL application. It does not really has the NVFlare job concept, or job scheduling. To enforce the min_clients requirement to run the simulator seems to be too strict.

Reply:
Right. We should not enforce the job deployment criteria here.